src/torch_ucc.cpp

```diff
@@ -596,18 +610,18 @@ ProcessGroupUCC::~ProcessGroupUCC() {
   if (comm) {
     comm->ucc_destroy_team(team);
     logger->logInfo(TORCH_UCC_FINALIZE, "Successfully destoryed UCC library");
-    comm->ucx_disconnect_eps(eps, &oob);
+    comm->ucx_disconnect_eps(eps, oob);
     logger->logInfo(TORCH_UCC_FINALIZE, "Successfully destoryed UCX library");
 		try {
     	if (cuda_ee) {
       	ucc_ee_destroy(cuda_ee);
     	}
-    	if ((size_t)oob.store->add(oob.getKey("ucc_pg_closed"), 1) == eps.size()) {
-				std::vector<uint8_t> val = {1};
-      	oob.store->set(oob.getKey("ucc_pg_finished"), val);
-    	} else {
-      	oob.store->wait({oob.getKey("ucc_pg_finished")});
-    	}
+      if ((size_t)oob->store->add(oob->getKey("ucc_pg_closed"), 1) == eps.size()) {
+        std::vector<uint8_t> val = {1};
+        oob->store->add(oob->getKey("ucc_pg_finished"), val);
```

Comment:
@Sergei-Lebedev - sorry I made a mistake when resolving the merge conflict and unable to push the fix here.
```suggestion
        oob->store->set(oob->getKey("ucc_pg_finished"), val);
```

Reply:
fixed